web/packages/teleport/src/BotInstances/List/BotInstancesList.tsx

```diff
@@ -0,0 +1,137 @@
+ */
+
+import formatDistanceToNowStrict from 'date-fns/formatDistanceToNowStrict';
+import styled from 'styled-components';
+
+import { Info } from 'design/Alert/Alert';
+import { Cell, LabelCell } from 'design/DataTable/Cells';
+import Flex from 'design/Flex';
+import Text from 'design/Text';
+import { CopyButton } from 'shared/components/UnifiedResources/shared/CopyButton';
+
+import { ClientSearcheableTableWithQueryParamSupport } from 'teleport/components/ClientSearcheableTableWithQueryParamSupport/ClientSearcheableTableWithQueryParamSupport';
+import { BotInstanceSummary } from 'teleport/services/bot/types';
+
+const MonoText = styled(Text)`
+  font-family: ${({ theme }) => theme.fonts.mono};
+`;
+
+export function BotInstancesList({
+  data,
+  pageSize = 20,
+}: {
+  data: BotInstanceSummary[];
+  pageSize?: number;
+}) {
+  const tableData = data.map(x => ({
+    ...x,
+    hostnameDisplay: x.host_name_latest ?? '-',
+    shortInstanceId: x.instance_id.substring(0, 7),
+    versionSortBy: x.version_latest ? semverExpand(x.version_latest) : 'Z',
+    versionDisplay: x.version_latest ? `v${x.version_latest}` : '-',
+    activeAtDisplay: x.active_at_latest
+      ? `${formatDistanceToNowStrict(new Date(x.active_at_latest))} ago`
+      : '-',
+  }));
+
+  return (
+    <ClientSearcheableTableWithQueryParamSupport
+      data={tableData}
+      columns={[
+        {
+          key: 'bot_name',
+          headerText: 'Bot',
+          isSortable: true,
+        },
+        {
+          key: 'instance_id',
+          isNonRender: true,
+        },
+        {
+          key: 'shortInstanceId',
+          headerText: 'ID',
+          isSortable: false,
+          render: ({ instance_id, shortInstanceId }) => (
+            <Cell>
+              <Flex inline alignItems={'center'} gap={1} mr={0}>
+                <MonoText>{shortInstanceId}</MonoText>
+                <CopyButton name={instance_id} />
+              </Flex>
+            </Cell>
+          ),
+        },
+        {
+          key: 'join_method_latest',
+          headerText: 'Method',
+          isSortable: true,
+          render: ({ join_method_latest }) =>
+            join_method_latest ? (
+              <LabelCell data={[join_method_latest]} />
+            ) : (
+              <Cell>{'-'}</Cell>
+            ),
+        },
+        {
+          key: 'hostnameDisplay',
+          headerText: 'Hostname',
+          isSortable: true,
+        },
+        {
+          key: 'versionDisplay',
+          headerText: 'Version (tbot)',
+          isSortable: true,
+          altSortKey: 'versionSortBy',
+        },
+        {
+          key: 'activeAtDisplay',
+          headerText: 'Last active',
+          isSortable: true,
+          altSortKey: 'active_at_latest',
+        },
+      ]}
+      emptyText="No active instances found"
+      emptyButton={
+        <Info alignItems="flex-start" mt={5}>
+          Bot instances are ephemeral, and disappear once their outputs (jwts,
```

Comment:
Perhaps simpler to shorten this copy to something like:

```suggestion
          Bot instances are ephemeral, and disappear once all issued credentials have expired.
```

Reply:
Amended.